src/local-run.ts

```diff
@@ -19,47 +16,26 @@ export const runLocally = async function () {
   }
   const payload = await ParsePayload(readAll);
 
-  // Only supports function_executed events
-  // To add support for other events, we need to support a way to provide function-centric handlers for them
-  // To support block_actions we would need a corresponding function callback_id provided for any events received
-  // in order to route it to the right function's actions handler
-  const eventType = payload?.body?.event?.type || payload?.body?.type || "";
-  if (eventType != "function_executed") {
-    throw new Error(`Unsupported event type: "${eventType || "not_found"}"`);
-  }
+  // Finds the corresponding function in the manifest definition, and then uses
+  // the `source_file` property to determine the function module file location
+  const resp = await DispatchPayload(payload, (functionCallbackId) => {
+    const functionDefn = manifest.functions[functionCallbackId];
+    if (!functionDefn) {
+      throw new Error(
+        `No function definition for function callback id ${functionCallbackId} was found in the manifest! manifest.functions: ${manifest.functions}`,
+      );
+    }
 
-  // Based on the function callback_id, look for a source_file property in the underlying manifest's function definition for local run
-  const functionCallbackId = payload?.body?.event?.function?.callback_id;
-  const functionDefn = manifest.functions[functionCallbackId];
-  if (!functionDefn) {
-    throw new Error(
-      `No function definition for function callback id ${functionCallbackId} was found in the manifest! manifest.functions: ${manifest.functions}`,
-    );
-  }
+    const functionFile =
+      `file://${workingDirectory}/${functionDefn.source_file}`;
 
-  const { dir: sourceDir, name: sourceFilename } = parse(
-    functionDefn.source_file,
-  );
-  const functionDir = `file://${workingDirectory}/${sourceDir}`;
-
-  if (sourceFilename !== payload?.body?.event?.function.callback_id) {
-    // Override the callback_id to point at the local file name
-    payload.body.event.function.callback_id = sourceFilename;
-  }
-
-  const functionModule = await LoadFunctionModule(
-    functionDir,
-    payload as InvocationPayload<FunctionInvocationBody>,
-  );
+    return [functionFile];
+  });
 
-  await RunFunction(
-    payload as InvocationPayload<FunctionInvocationBody>,
-    functionModule,
-  );
   // The CLI expects a JSON payload to be output to stdout
   // This is formalized in the `run` hook of the CLI/SDK Tech Spec:
   // https://corp.quip.com/0gDvAsqoaaYE/Proposal-CLI-SDK-Interface#temp:C:fOC1991c5aec8994d0db01d26260
-  console.log("{}");
+  console.log(JSON.stringify(resp || {}));
```

Comment:
Yah, the CLI uses this output as the response to the HTTP request that delivered the event/payload. It will let some interactivity handlers, like view_submission, to do things like return validation error responses.